src/main/java/com/minecolonies/coremod/colony/crafting/CustomRecipeManager.java

```diff
@@ -85,6 +89,19 @@ public void removeRecipe(@NotNull final JsonObject recipeJson, @NotNull final Re
      */
     public Set<CustomRecipe> getRecipes(@NotNull final String crafter)
     {
+        if(!removedRecipes.isEmpty())
+        {
+            for(ResourceLocation toRemove: removedRecipes)
+            {
+                final Optional<HashMap<ResourceLocation, CustomRecipe>> crafterMap = recipeMap.entrySet().stream().map(r -> r.getValue()).filter(r1 -> r1.keySet().contains(toRemove)).findFirst();
```

Comment:
isnt that the same as recipeMap.get(toRemove)?

Reply:
No, recipeMap is a <String, Map<ResourceLocation, CustomRecipe>> where the string is the crafter name, since that is primary access route. During removal, I don't know which crafter it's coming from, so we're retrieving the correct submap.

Reply:
ah true, had assumed its a resloc map